jolt-core/src/host/toolchain.rs

```diff
@@ -22,13 +25,40 @@ pub fn install_toolchain() -> Result<()> {
     let toolchain_url = toolchain_url();
 
     let rt = Runtime::new().unwrap();
-    rt.block_on(download_toolchain(&client, &toolchain_url))?;
+    println!("Downloading toolchain (retrying up to {DOWNLOAD_RETRIES} times)");
+    rt.block_on(retry_times(DOWNLOAD_RETRIES, DELAY_BASE_MS, || {
+        download_toolchain(&client, &toolchain_url)
+    }))?;
     unpack_toolchain()?;
     link_toolchain()?;
 
     write_tag_file()
 }
 
+async fn retry_times<F, T, E>(times: usize, base_ms: u64, f: F) -> Result<T>
+    where
+        F: Fn() -> E,
+        E: Future<Output=Result<T>>,
+{
+    for i in 0..times {
+        println!("Attempt {}/{}", i + 1, times);
+        match f().await {
+            Ok(t) => return Ok(t),
+            Err(e) => {
+                let timeout = delay_timeout(i, base_ms);
+                println!("Error: {}. Retrying in {}ms", e, timeout);
```

Comment:
nit: `println!("Toolchain download error {i}/{times}: {e}. Retrying in {timeout}ms")`

Reply:
Done